Modules/CustomerHistory/CustomerHistory.lua

```diff
@@ -34,13 +34,14 @@ function CraftSim.CUSTOMER_HISTORY:HandleWhisper(event, message, customer, ...)
     while (table.getn(self.db.realm[customer].history) > CraftSimOptions.maxHistoryEntriesPerClient) do
         table.remove(self.db.realm[customer].history, 1)
     end
-    CraftSim.CUSTOMER_HISTORY.FRAMES:AddCustomer(customer)
     CraftSim.CUSTOMER_HISTORY.FRAMES:SetCustomer(customer)
 end
 
-function CraftSim.CUSTOMER_HISTORY:LoadMessages()
-    CraftSim.CUSTOMER_HISTORY.FRAMES:SetCustomer()
-    self:UnregisterEvent("TRADE_SKILL_SHOW")
+function CraftSim.CUSTOMER_HISTORY:LoadHistory()
+    if not self.db.realm.lastCustomer then
+        CraftSim.CUSTOMER_HISTORY.FRAMES:ResetDropdown()
```

Comment:
I am not sure to understand why this call is needed, mind explaining?

Reply:
So what happened was : let's say I /reload and `lastCustomer` is nil, when opening the tradeskill window, the dropdown would be empty because calling `SetCustomer` with a nil customer returns, and thus does not reload the dropdown.

But you're right, this is dumb. I'll just move the return condition in SetCustomer to LoadHistory because it's the only function where customer can't be nil

Reply:
Also, ResetDropdown doesn't empty the dropdown, it resets it to a list of customer (partial if a filter is provided)

Reply:
It would be empty but filled as soon as you click on it/search for something, right? Nothing would be selected anyway since  `lastCustomer` is nil, and the user can't notice that it is empty. At least that's what I think?

Reply:
It is actually loaded because of `OnTextChanged` being called when the filter box is initialized.

Reply:
Nah that’s the thing, it wouldn’t have loaded the dropdown at all. Both nothing selected and an empty list. But it should be fine with the recent changes I made

Reply:
> It is actually loaded because of `OnTextChanged` being fired when the filter box is initialized.

Ah yeah, but if you were to just open the drop-down it would’ve be empty. Anyway that’s all in the past now